citadel_sdk/src/remote_ext.rs

```diff
@@ -548,80 +640,133 @@ pub trait ProtocolRemoteTargetExt: TargetLockedRemote {
+                                                 }) => {
+                    let mut local_path = None;
+                    while let Some(res) = handle.next().await {
+                        match res {
+                            ObjectTransferStatus::ReceptionBeginning(path, _) => {
+                                local_path = Some(path)
+                            }
+                            ObjectTransferStatus::TransferComplete => {
+                                break;
+                            }
+                            _ => {}
+                        }
+                    }
+                    if local_path.is_some() {
+                        Ok(local_path)
+                    } else {
+                        Err(NetworkError::InternalError("Local path never loaded"))
                     }
                 }
 
-                Ok(local_path.ok_or(NetworkError::InternalError("Local path never loaded"))?)
-            }
-
-            res => {
-                log::error!(target: "citadel", "Invalid NodeResult for FileTransfer request received: {:?}", res);
-                Err(NetworkError::InternalError(
-                    "Received invalid response from protocol",
-                ))
+                res => {
+                    log::error!(target: "citadel", "Invalid NodeResult for FileTransfer request received: {:?}", res);
+                    Err(NetworkError::InternalError(
+                        "Invalid Response From Protocol",
+                    ))
+                }
             }
-        }
+        }).await
     }
 
-    /// Deletes the file from the RE-VFS. If the contents are desired on delete,
-    /// consider calling `Self::remote_encrypted_virtual_filesystem_pull` with the delete
-    /// parameter set to true
-    async fn remote_encrypted_virtual_filesystem_delete<R: Into<PathBuf> + Send>(
+    /// Deletes the file from the RE-VFS. If the contents are desired on delete, consider calling
+    /// `Self::remote_encrypted_virtual_filesystem_pull` with the delete parameter set to true.
+    /// Uses a custom function to handle the response. The function should return Some if the
+    /// signal is the required signal needed to terminate the process, None otherwise.
+    async fn remote_encrypted_virtual_filesystem_delete_with_fn<R, F, T>(
         &self,
         virtual_directory: R,
-    ) -> Result<(), NetworkError> {
+        mut f: F,
+    ) -> Result<(), NetworkError>
+    where
+        R: Into<PathBuf> + Send,
+        F: FnMut(NodeResult) -> T + Send,
+        T: Future<Output = Result<(), NetworkError>> + Send,
+    {
         self.can_use_revfs()?;
         let request = NodeRequest::DeleteObject(DeleteObject {
             v_conn: *self.user(),
             virtual_dir: virtual_directory.into(),
             security_level: Default::default(),
         });
+        let status = self.remote().send_callback(request).await?;
```

Comment:
Recall that our issue is that when peers are behind the same kernel, there can be false positive matches when the tickets match. We also needed to check CIDs. The `send_callback` and `send_callback_subscription` functions only check for ticket matching; `send_callback` only waits for the first signal to come back, whereas `send_callback_subscription` streams signals until a signal we want comes through. Since we now require the possibility of multiple signals, we will need to upgrade `send_callback` to `send_callback_subscription`, since in the IS, we may get a ticket match but not a CID match and will need to reroute that signal, then keep waiting. So, everywhere that we do a `send_callback`, we will need to upgrade it to the streaming version.